handlers/sf-move-subscriptions-api/src/main/scala/com/gu/sf/move/subscriptions/api/UpdateSupporterProductDataService.scala

```diff
@@ -0,0 +1,59 @@
+package com.gu.sf.move.subscriptions.api
+import cats.implicits.toTraverseOps
+import com.gu.effects.sqs.AwsSQSSend.{Payload, QueueName}
+import com.gu.effects.sqs.{AwsSQSSend, SqsSync}
+import com.gu.zuora.subscription.Subscription
+import com.typesafe.scalalogging.LazyLogging
+import io.circe.syntax.EncoderOps
+
+trait UpdateSupporterProductData {
+  def update(subscription: Subscription, identityId: String): Either[String, Unit]
+}
+
+class UpdateSupporterProductDataService(queueName: QueueName) extends UpdateSupporterProductData with LazyLogging {
+
+  def combineErrorsOrUnit(list: List[Either[String, Unit]]): Either[String, Unit] = {
+    list.foldLeft[Either[String, Unit]](Right(())) { (acc, either) =>
+      acc match {
+        case Left(errors) =>
+          either match {
+            case Left(error) => Left(s"$errors, $error")
+            case Right(_) => acc
+          }
+        case Right(_) =>
+          either match {
+            case Left(error) => Left(s"$error")
+            case Right(_) => acc
+          }
+      }
+    }
```

Comment:
I assume you're trying to avoid cats/implicits here -
haven't compiled but something like this would be more in line with the intention
```suggestion
    val errors = list.collect { case Left(error) => error }
    if (errors.isEmpty) Right(()) else Left(errors.mkString(", "))
```